From e2698477e8abf623c18ab28d33cc894ec882a706 Mon Sep 17 00:00:00 2001
From: Neil Horman <neil.horman@privafy.com>
Date: Fri, 18 Mar 2022 18:59:52 -0400
Subject: [PATCH] Adjust jitterentropy detection to look for the settick
 function

Theres no great way to detect if jitterentropy has the internal timer
feature enabled so we have to look for a function that is only defined
when it is enabled
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index e16e1a0..0f5a38e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -94,7 +94,7 @@ AS_IF(
 		AC_SEARCH_LIBS(jent_version,jitterentropy,
 				[AM_CONDITIONAL([JITTER], [true])
 				AC_DEFINE([HAVE_JITTER],1,[Enable JITTER])
-				AC_CHECK_LIB(jitterentropy, jent_entropy_switch_notime_impl,
+				AC_CHECK_LIB(jitterentropy, jent_notime_settick,
 				[AC_DEFINE([HAVE_JITTER_NOTIME],1,[Enable JITTER_NOTIME])],
 				[],-lpthread)],
 				AC_MSG_NOTICE([No Jitterentropy library found]),-lpthread)
